nupic/research/frameworks/vernon/mixins/oml.py

```diff
@@ -173,3 +280,31 @@ def get_execution_order(cls):
         eo["create_loaders"].append("Create loaders for the meta testing phase")
 
         return eo
+
+
+class ResetOMLTaskParams(object):
```

Comment:
can you explain why we need to make this a separate mixin, rather than just defining the `pre_task` method used here in the OMLExperiment class itself?

Reply:
Fair point. As of now `ResetOMLTaskParams` only works with the `OMLNetwork` whereas the `OnlineMetaLearning` could technically work for any network and is thus broader in scope.

If there was a generalized way to reset the parameters as needed, then I think we could incorporate it into `OnlineMetaLearning`, but I'm not sure that's the case. For instance, even if it always involved resetting the weights in the last linear layer, how will we know how to access those weights? They typically will be `list(self.model.parameters())[-2]`, but that will not necessarily always be true.

We could also enforce the model to have an `output_params` attribute, but I didn't want to enforce more constraints past the `named_fast_params` and `named_slow_params`. At any rate, I figure the best solution will come with further experiments and further understanding our needs.